### PR TITLE
fix AbstractController#action_methods 

### PR DESCRIPTION
**Pull request by kamil-gwozdz**
This PR fixes `AbstractController#action_methods` to include inherited shadowed internal methods.

### Motivation / Background

I tried enabling [`raise_on_missing_callback_actions`](https://github.com/rails/rails/pull/43487) and I found out that `MyController#action_methods` is missing a method that `MyController` actually defines. It only happens when a method is also defined by an ancestor abstract controller.

### Checklist

Before submitting the PR make sure the following are checked:

* [ ] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [ ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: actionpack/lib/action_dispatch/routing/route_set.rb [hunk 1]
```diff
@@ -579,6 +580,8 @@ def url_options; {}; end
             extend path_helpers
           end
 
+          extend AbstractController::IgnoreActionMethods
```
Comment (kamil-gwozdz) [inline actionpack/lib/action_dispatch/routing/route_set.rb:583]: `_routes` is defined in that module, which is included directly in the controller. It's a public method but we don't want to include it in `action_methods`.

**Comment by Edouard-chin** [comment]
Thanks for the PR  ❤️ !

What do you think about tweaking the `internal_methods` method instead? Because I think the underlying problem is there (in your test case if you call `ChildController.internal_methods`, we end up with a `status` which is not correct).

I think it will make the fix easier and would avoid the `IgnoreActionMethods` module workaround.

**Comment by kamil-gwozdz** [comment]
> in your test case if you call ChildController.internal_methods, we end up with a status which is not correct

@Edouard-chin my impression was that `internal_methods` is actually correct because `status` is defined here: https://github.com/rails/rails/blob/39156c8812738bf7bef1ec749c84b4e495ffb3df/actionpack/lib/action_controller/metal.rb#L183

**Comment by matthewd** [comment]
I don't think `internal_methods` has value outside its current use in `action_methods`; we can probably just deprecate and remove it. In the meantime, I think we can amend it to subtract `action_methods`.

**Comment by kamil-gwozdz** [comment]
https://github.com/rails/rails/pull/48699

**Pull request closed without merging**